libs/sdk-ui/src/base/results/internal/resultDataMethods.ts

```diff
@@ -122,6 +151,38 @@ class ResultDataMethods implements IResultDataMethods {
         return this.dataView.totals;
     }
 
+    public rowTotals(): DataValue[][] | undefined {
+        return this.dataView.totals?.[0];
+    }
+
+    public columnTotals(): DataValue[][] | undefined {
+        return this.dataView.totals?.[1];
+    }
+
+    public totalOfTotals(): DataValue[][][] | undefined {
+        return this.dataView.totalTotals;
+    }
+
+    /**
+     *
+     * @returns - existance of totals for rows
+     */
+    public hasRowTotals(): boolean {
+        return this.dataView.totals?.[0] !== undefined && this.dataView.totals[0].length > 0;
+    }
+
+    /**
+     *
+     * @returns - existance of totals for columns
+     */
+    public hasColumnTotals(): boolean {
+        return this.dataView.totals?.[1] !== undefined && this.dataView.totals[1].length > 0;
```

Comment:
what about to reuse this.columnTotals() internally?

Reply:
typescript validation as discussed